include/aspect/compat.h

```diff
@@ -83,6 +83,67 @@ namespace dealii
 }
 #endif
 
+/*
+ * Unique_ptr functionality that was introduced in deal.II 8.3 and replaces
+ * auto_ptr in CXX17. We need this to silence deprecation warnings in new
+ * compilers.
+ */
+#if !DEAL_II_VERSION_GTE(8,3,0)
```

Comment:
So is it a good idea to depend on this fact? I would probably write the code as:
```
#if DEAL_II_VERSION_GTE(8,3,0)
#include <deal.II/base/std_cxx11/unique_ptr.h>
#else
namespace dealii
{
  namespace std_cxx11
  {
...
```
This is not only a lot more clear, it is also more robust.